source/fundamentals/aggregation.txt

```diff
@@ -7,51 +9,232 @@ Aggregation
+       | Sort the results
+       | Limit the results
+       | Count the results
+       | Rename fields
+       | Calculate fields
+       | Summarize data
+       | Group values
+
+Limitations
+-----------
+
+Aggregation operations have limitations. When performing agregation
+operations, keep the following in mind:
+
+- Returned documents must not violate the :manual:`BSON document size
+  limit </reference/limits/#BSON-Document-Size>` of 16 megabytes.
+- Pipeline stages have a memory limit of 100 megabytes by default. If
+  required, you may exceed this limit by using the `allowDiskUse
+  <{+api+}/mongo/options#AggregateOptions.SetAllowDiskUse>`__
+  method.
+- The :manual:`$graphLookup
+  </reference/operator/aggregation/graphLookup/>` stage
+  has a strict memory limit of 100 megabytes and ignores
+  ``allowDiskUse``.
+
+Examples
+--------
 
-     The :manual:`$graphLookup </operator/aggregation/graphLookup/>` stage has a strict memory limit of 100 megabytes
-     and will ignore ``allowDiskUse``.
+To run the examples in this section, load the sample data into the
+``tea.menu`` collection with the following snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/aggregation.go
+   :start-after: begin insert docs
+   :end-before: end insert docs
+   :language: go
+   :dedent:
+
+Each document represents a tea on the menu of a shop. It conatains
+information about the tea type, the toppings you can add, and its price.
+
+Average Rating
+~~~~~~~~~~~~~~
+
+The following example calculates and displays the average rating and
+amount of ratings for each tea.
+
+The aggregation pipleline uses the ``$group`` stage to group the
+documents by the ``category`` field, calculates the average using the
+``$avg`` expression operator, and counts the number of documents using
+the ``$sum`` expression operator.
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: go
+
+      // create the stage
+      groupStage := bson.D{
+          {"$group", bson.D{
+              {"_id", "$category"},
+              {"average_price", bson.D{
+                  {"$avg", "$price"},
+              }},
+              {"type_total", bson.D{
+                  {"$sum", 1},
+              }},
+      }}}
+
+      // pass the stage into a pipeline
+      // pass the pipeline as the second paramter in the Aggregate() method
+      cursor, err := coll.Aggregate(context.TODO(), mongo.Pipeline{groupStage})
+      if err != nil {
+          panic(err)
+      }
+
+      // display the results
+      var results []bson.M
+      if err = cursor.All(context.TODO(), &results); err != nil {
+          panic(err)
+      }
+      for _, result := range results {
+          fmt.Printf("%v tea has an average price of $%v \n", result["_id"], result["average_price"])
+          fmt.Printf("Amount of %v tea: %v \n\n", result["_id"], result["type_total"])
+      }
+
+   .. output::
+      :language: go
+
+      Average price of black tea: $6.075 
```

Comment:
This would probably be `black tea has an average price of $6.075` given the print statement on L142. Not sure if that matters, though.

Reply:
Ah yes definitely matters! I forgot to update the print statement -- good catch!